### DISCUSSION TITLE
How can I add a precompiled library to vcpkg?

### DISCUSSION THREAD

**Discussion by serg06** [microsoft/vcpkg#18594] [Q&A] [answered]
To preface, I have the following library in a private Github repository:

```
cef/
    include/
        ...
    precompiled/
        Release/
            libcef.lib
            libcef_dll_wrapper.lib
```

I want to use it in my project. The project doesn't use CMake, so I `git clone` the library then add it to the `.vcxproj`:

```
<AdditionalIncludeDirectories>
    ...
    $(CefLib)\include
<AdditionalIncludeDirectories/>

<AdditionalDependencies>
    ...
    $(CefLib)\precompiled\Release\libcef.lib;
    $(CefLib)\precompiled\Release\libcef_dll_wrapper.lib
<AdditionalDependencies/>
```

Is there any way to make this library compatible with vcpkg? I can't figure out what to put in the library's `CMakeLists.txt`.

**Comment by autoantwort** [comment, marked as answer]
I think you can use [this](https://github.com/microsoft/vcpkg/blob/master/ports/chartdir/portfile.cmake) as reference

**Reply by serg06** [reply]
Thank you, this solved so many problems for me. 

TL;DR for future readers:
- Copy the `.lib` files to `${CURRENT_PACKAGES_DIR}/lib`
- Copy the `.dll` files to `${CURRENT_PACKAGES_DIR}/dll`
- Copy the includes to `${CURRENT_PACKAGES_DIR}/include`
- Copy the license to `${CURRENT_PACKAGES_DIR}/copyright`

And everything just works.

(Well... mostly. I had to do a few more tweaks to fix `libcef_dll_wrapper` linker issues and vcpkg's post-build "Could not find proper second linker member" error.)

**Comment by nnalen595-code** [comment]
#49597 
[607704ba-aaf2-4946-82d8-c2734063ae70.zip](https://github.com/user-attachments/files/24845837/607704ba-aaf2-4946-82d8-c2734063ae70.zip)
[``]([url]([url](**### url**)))
